pandas/_libs/tslibs/strptime.pyx

```diff
@@ -129,12 +130,22 @@ def array_strptime(ndarray[object] values, str fmt, bint exact=True, errors='rai
             if val in nat_strings:
                 iresult[i] = NPY_NAT
                 continue
+        elif checknull_with_nat_and_na(val):
+            iresult[i] = NPY_NAT
+            continue
+        elif isinstance(val, datetime):
+            iresult[i] = _parse_python_datetime_object(val, &dts)
```

Comment:
Instead of this could we just do:

```
val = val.strftime(fmt)
```

And then the existing parsing code can be reused?

Reply:
This does not work because `iresult` contains int64 date representations of the dates. IIRC, those values are used to represent the `Timestamp` objects internally

Reply:
Right but I mean here we convert the `datetime` to a `str` and the `for group_key in found_dict.iterkeys():` loop below will assign the `iresult[i]` value of the `datetime -> str`

Reply:
I see it now. That works too. Seems like it depends on deciding if `datetime` and `Timestamp` objects should be left alone or formatted.

For example:

```python
>>> data = [Timestamp("2001-10-02 12:30:01"), "01/10/01"]
>>> 
>>> # Current approach: leave Timestamp/datetime objects alone
>>> pd.to_datetime(data, format="%d/%m/%y")
DatetimeIndex(['2001-10-02 12:30:01', '2001-10-01 00:00:00'], dtype='datetime64[ns]', freq=None)
>>> 
>>> # Using val = val.strftime(fmt): apply format on Timestamp/datetime objects
>>> pd.to_datetime(data, format="%d/%m/%y")
DatetimeIndex(['2001-10-02', '2001-10-01'], dtype='datetime64[ns]', freq=None)
```

I thought that leaving the objects alone (skipping them) was cleaner, but seems like it produces unexpected results. Applying `fmt` on `Timestamp` and `datetime` objects seems like a better approach now. Thanks for your suggestion!